m1 Mac Build: variable set but not used

### Description

It is currently not possible to build the mumble client on macOS on ARM.

### Steps to reproduce

```
cd build
cmake -Dice=off -DCMAKE_BUILD_TYPE=Release ..
cmake --build . -j 4
```

### Mumble version

1.4.x

### Mumble component

Client

### OS

macOS

### Reproducible?

Yes

### Additional information

The following dummy "uses" `cgimg` but probably isn't the nicest solution

```diff
diff --git a/src/mumble/Overlay_macx.mm b/src/mumble/Overlay_macx.mm
index ae884dc98..e89ec6c1d 100644
--- a/src/mumble/Overlay_macx.mm
+++ b/src/mumble/Overlay_macx.mm
@@ -213,6 +213,7 @@ void OverlayClient::updateMouse() {
 				cgimg = [(NSBitmapImageRep *)rep CGImage];
 			}
 		}
+    (void) cgimg;
 	}
 
 	NSPoint p = [cursor hotSpot];
```

### Relevant log output

```shell
[ 99%] Building CXX object src/mumble/CMakeFiles/mumble.dir/PAAudio.cpp.o
/Users/thegcat/Code/m1/mumble/src/mumble/Overlay_macx.mm:208:14: error: variable 'cgimg' set but not used [-Werror,-Wunused-but-set-variable]
                CGImageRef cgimg = nullptr;
                           ^
1 error generated.
make[2]: *** [src/mumble/CMakeFiles/mumble.dir/Overlay_macx.mm.o] Error 1
```


### Screenshots

_No response_

--- a/src/mumble/Overlay_macx.mm
+++ b/src/mumble/Overlay_macx.mm
@@ -203,17 +203,6 @@ void OverlayClient::updateMouse() {
 	}
 
 	QPixmap pm = qmCursors.value(csShape);
-	if (pm.isNull()) {
-		NSImage *img = [cursor image];
-		CGImageRef cgimg = nullptr;
-		NSArray *reps = [img representations];
-		for (NSUInteger i = 0; i < [reps count]; i++) {
-			NSImageRep *rep = [reps objectAtIndex:i];
-			if ([rep class] == [NSBitmapImageRep class]) {
-				cgimg = [(NSBitmapImageRep *)rep CGImage];
-			}
-		}
-	}
 
 	NSPoint p = [cursor hotSpot];
 	iOffsetX = (int) p.x;

